test/integration/throttleTest.ts

```diff
@@ -0,0 +1,69 @@
+import { strict as assert } from "assert";
+import { newTestUser, overrideRatelimitForUser, resetRatelimitForUser } from "./clientHelper";
+import { getMessagesByUserIn } from "../../src/utils";
+import { onReactionTo } from "./commands/commandUtils";
+
+describe("Test: throttled users can function with Mjolnir.", function () {
+    it('throttled users survive being throttled by synapse', async function() {
+        this.timeout(60000);
+        let throttledUser = await newTestUser({ name: { contains: "throttled" }, isThrottled: true });
+        let throttledUserId = await throttledUser.getUserId();
+        let targetRoom = await throttledUser.createRoom();
+        // send enough messages to hit the rate limit.
+        await Promise.all([...Array(150).keys()].map((i) => throttledUser.sendMessage(targetRoom, {msgtype: 'm.text.', body: `Message #${i}`})));
+        let messageCount = 0;
+        await getMessagesByUserIn(throttledUser, throttledUserId, targetRoom, 150, (events) => {
+            messageCount += events.length;
+        });
+        assert.equal(messageCount, 150, "There should have been 150 messages in this room");
+    })
+})
+
+describe("Test: Mjolnir can still sync and respond to commands while throttled", function () {
+    beforeEach(async function() {
+        await resetRatelimitForUser(await this.mjolnir.client.getUserId())
+    })
+    afterEach(async function() {
+        await overrideRatelimitForUser(await this.mjolnir.client.getUserId());
+    })
+
+    it('Can still perform and respond to a redaction command', async function () {
+        this.timeout(60000);
+        // Create a few users and a room.
+        let badUser = await newTestUser({ name: { contains: "spammer-needs-redacting" } });
+        let badUserId = await badUser.getUserId();
+        const mjolnir = this.mjolnir.client;
+        let mjolnirUserId = await mjolnir.getUserId();
+        let moderator = await newTestUser({ name: { contains: "moderator" } });
+        this.moderator = moderator;
+        await moderator.joinRoom(this.mjolnir.managementRoomId);
+        let targetRoom = await moderator.createRoom({ invite: [await badUser.getUserId(), mjolnirUserId]});
+        await moderator.setUserPowerLevel(mjolnirUserId, targetRoom, 100);
+        await badUser.joinRoom(targetRoom);
+        moderator.sendMessage(this.mjolnir.managementRoomId, {msgtype: 'm.text.', body: `!mjolnir rooms add ${targetRoom}`});
```

Comment:
I think that we should also have the moderator send many messages, just to be certain that we trigger the rate limit.